src/ProxyManager/GeneratorStrategy/FileWriterGeneratorStrategy.php

```diff
@@ -55,13 +63,43 @@ public function generate(ClassGenerator $classGenerator)
             . '\\' . trim($classGenerator->getName(), '\\');
         $generatedCode = $classGenerator->generate();
         $fileName      = $this->fileLocator->getProxyFileName($className);
-        $tmpFileName   = $fileName . '.' . uniqid('', true);
 
-        // renaming files is necessary to avoid race conditions when the same file is written multiple times
-        // in a short time period
-        file_put_contents($tmpFileName, "<?php\n\n" . $generatedCode);
-        rename($tmpFileName, $fileName);
+        set_error_handler($this->emptyErrorHandler);
+
+        try {
+            $this->writeFile("<?php\n\n" . $generatedCode, $fileName);
+        } catch (FileNotWritableException $fileNotWritable) {
+            restore_error_handler();
+
+            throw $fileNotWritable;
+        }
+
+        restore_error_handler();
```

Comment:
Thanks, will fix
On Aug 9, 2015 23:06, "Jefersson Nathan" <notifications@github.com> wrote:

> In src/ProxyManager/GeneratorStrategy/FileWriterGeneratorStrategy.php
> <https://github.com/Ocramius/ProxyManager/pull/251#discussion_r36594255>:
>
> >
> > -        // renaming files is necessary to avoid race conditions when the same file is written multiple times
> > -        // in a short time period
> > -        file_put_contents($tmpFileName, "<?php\n\n" . $generatedCode);
> > -        rename($tmpFileName, $fileName);
> > +        set_error_handler($this->emptyErrorHandler);
> > +
> > +        try {
> > +            $this->writeFile("<?php\n\n" . $generatedCode, $fileName);
> > +        } catch (FileNotWritableException $fileNotWritable) {
> > +            restore_error_handler();
> > +
> > +            throw $fileNotWritable;
> > +        }
> > +
> > +        restore_error_handler();
>
> Cannot be?
>
> try {    $this->writeFile("<?php\n\n" . $generatedCode, $fileName);} catch (FileNotWritableException $fileNotWritable) {    throw $fileNotWritable;} finally {    restore_error_handler();}
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ocramius/ProxyManager/pull/251/files#r36594255>.
>